### PR TITLE
3rdparty

### PR DESCRIPTION
**Pull request by djmitche**
If you like this, I'll email a few of the folks I've talked to about the topic to point them to the result.

### CODE REVIEW

File: src/manual/3rdparty/index.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+*Do not* use clientIds for authentication.
+First, because mere posession of a credential with a clientId and some 44-character accessToken does not prove anything until you have validated that the accessToken is valid.
+Second, even if you validate the accessToken, TaskCluster is fairly permissive in creation of temporary credentials with arbitrary clientIds, by design.
+The information you may rely on for authorization is contained in the list of scopes returned from the [auth.authenticateHawk](/reference/platform/auth/api-docs#authenticateHawk) method.
+
+# Approaches
+
+## Frontend-Only Use
+
+If your application is a frontend-only tool, or only needs to call TaskCluster APIs on behalf of the user, you can perform the Access Grant process, then store the resulting credentials in the JavaScript heap or (to survive reloads) LocalStorage.
```
Comment (jonasfj) [inline src/manual/3rdparty/index.md]: `LocalStorage`

File: src/manual/3rdparty/index.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+Instead, treat the scopes returned from [auth.authenticateHawk](/reference/platform/auth/api-docs#authenticateHawk) as assertions of identity and group membership.
+
+In particular, look for scopes of the forms:
+
+ * `assume:mozilla-user:<email>`
+ * `assume:mozillians-user:<username>`
+ * `assume:mozilla-group:<group>`
+ * `assume:mozillians-group:<group>`
+
+This is one of the rare cases where you should *not* apply scope expansion; that is, posessing `assume:mozillians-user:*` should not allow a user to see data for all mozillians!
```
Comment (jonasfj) [inline src/manual/3rdparty/index.md]: I disagree!

**Comment by jonasfj** [comment]
# `r+`

Maybe add a section on trouble shooting or something...
Notably I people really often forget to deal with the `certificate` part of the temporary credentials... Inserting some sort of reminder to do this would be really nice...

File: src/manual/3rdparty/index.md (hunk 1 above)
Comment (djmitche) [inline src/manual/3rdparty/index.md]: That's what I said..?

File: src/manual/3rdparty/index.md (hunk 2 above)
Comment (djmitche) [inline src/manual/3rdparty/index.md]: For authorization, you should and do disagree.  For *authentication*, we do not want 3rd party services to accidentally grant us, the TC admins, godlike access to their services.

Comment (jonasfj) [inline src/manual/3rdparty/index.md]: Hmm, so you mean to say I should disregard my god-like aspirations...

I guess it's an acceptable argument to make...

Note:  thinking we should add to auth an end-point like authenticateHawk where you as the caller specifies both scopes and request signature... And auth just returns yes/no, including an error message if it is a no...

Comment (djmitche) [inline src/manual/3rdparty/index.md]: yeah, I like that idea

**Pull request merged**